dcp/api/job.py

```diff
@@ -167,14 +167,28 @@ def wait(self):
             return dry.aio.blockify(self._wait)()
 
         def on(self, *args):
+            # deserialize job on event parameters before passing them to user defined callback
+            def cb_deserialize_wrapper(callback):
+                def new_cb(*inner_args):
+                    nonlocal callback
+                    new_args = []
+                    for arg in inner_args:
+                        if isinstance(arg, dict) and bool(arg):
+                            for key in arg:
+                                arg[key] = deserialize(arg[key], self.serializers)
+                        new_args.append(deserialize(arg, self.serializers))
```

Comment:
Can a dict still be passed into the `deserialize` function?

Reply:
It would be a no-op in that case

I'm only deserializing the value in case user created data is returned at the top level (perhaps for custom events?)